### PR TITLE
wayland-proxy-virtwl: unstable-2023-08-13 -> unstable-2023-10-27

### PR DESCRIPTION
**Pull request by alyssais**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/wayland/default.nix [hunk 1]
```diff
@@ -2,28 +2,29 @@
...
 }:
 
 buildDunePackage rec {
   pname = "wayland";
-  version = "1.1";
+  version = "2.0";
 
-  minimalOCamlVersion = "4.08";
+  minimalOCamlVersion = "5.0";
   duneVersion = "3";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/wayland/default.nix:19]:
```suggestion
```

This line can be dropped.

File: pkgs/tools/wayland/wayland-proxy-virtwl/default.nix [hunk 2]
```diff
@@ -8,16 +8,16 @@
...
     repo = pname;
-    rev = "050c49a377808105b895e81e7e498f35cc151e58";
-    sha256 = "sha256-6YJv3CCED6LUSPFwYQyHUFkkvOWZGPNHVzw60b5F8+c=";
+    rev = "cc9548c4980ff33f86d5645ce337a79bf95d6139";
+    sha256 = "sha256-aAqbPslTu+RLQPKPJQH2iYjcI8/De2WPk5nHULdfocU=";
   };
 
-  minimalOCamlVersion = "4.12";
+  minimalOCamlVersion = "5.0";
   duneVersion = "3";
```
Comment (vbgl) [inline pkgs/tools/wayland/wayland-proxy-virtwl/default.nix:21]:
```suggestion
```

Same here.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -4796,7 +4796,9 @@ with pkgs;
 
   wayland-utils = callPackage ../tools/wayland/wayland-utils { };
 
-  wayland-proxy-virtwl = callPackage ../tools/wayland/wayland-proxy-virtwl { };
+  wayland-proxy-virtwl = callPackage ../tools/wayland/wayland-proxy-virtwl {
+    ocamlPackages = ocaml-ng.ocamlPackages_5_0;
```
Comment (vbgl) [inline pkgs/top-level/all-packages.nix:4800]: Why not OCaml 5.1, which is the latest available in `nixpkgs`?

Comment (alyssais) [reply pkgs/top-level/all-packages.nix:4800]: Because it doesn't work:

```
       error: ocaml-migrate-parsetree is not available for OCaml 5.1.0
```

Comment (vbgl) [reply pkgs/top-level/all-packages.nix:4800]: Sounds like a strong argument!

**Pull request merged**